packages/shared-ux/file/util/src/image_metadata.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import * as bh from 'blurhash';
+import type { FileImageMetadata } from '@kbn/shared-ux-file-image-types';
+
+export function isImage(file: { type?: string }): boolean {
+  return Boolean(file.type?.startsWith('image/'));
+}
+
+export const boxDimensions = {
+  width: 120,
+  height: 120,
+};
+
+/**
+ * Calculate the size of an image, fitting to our limits see {@link boxDimensions},
+ * while preserving the aspect ratio.
+ */
+export function fitToBox(width: number, height: number): { width: number; height: number } {
+  const offsetRatio = Math.abs(
+    Math.min(
+      // Find the aspect at which our box is smallest, if less than 1, it means we exceed the limits
+      Math.min(boxDimensions.width / width, boxDimensions.height / height),
+      // Values greater than 1 are within our limits
+      1
+    ) - 1 // Get the percentage we are exceeding. E.g., 0.3 - 1 = -0.7 means the image needs to shrink by 70% to fit
+  );
+  return {
+    width: Math.floor(width - offsetRatio * width),
+    height: Math.floor(height - offsetRatio * height),
+  };
+}
+
+/**
+ * Get the native size of the image
+ */
+function loadImage(src: string): Promise<HTMLImageElement> {
+  return new Promise((res, rej) => {
+    const image = new window.Image();
+    image.src = src;
+    image.onload = () => res(image);
+    image.onerror = rej;
+  });
+}
+
+/**
+ * Extract image metadata, assumes that file or blob as an image!
+ */
+export async function getImageMetadata(file: File | Blob): Promise<undefined | FileImageMetadata> {
+  const imgUrl = window.URL.createObjectURL(file);
+  try {
+    const image = await loadImage(imgUrl);
+    const canvas = document.createElement('canvas');
+    // blurhash encoding and decoding is an expensive algorithm,
+    // so we have to shrink the image to speed up the calculation
+    const { width, height } = fitToBox(image.width, image.height);
+    canvas.width = width;
+    canvas.height = height;
+    const ctx = canvas.getContext('2d');
+    if (!ctx) throw new Error('Could not get 2d canvas context!');
```

Comment:
we are throwing an error here and then catching it further below and swallowing it. should we log this somewhere?

Reply:
Good point!

Generating a blurhash for an image is kind of a nice-to-have. If it fails for whatever reason we definitely do not want to block the user. However, swallowing this error entirely might hide cases where blurhash generation fails (for whatever reason). It'd be great to record failures somehow, perhaps to our telemetry cluster (a simple UI counter could be useful?). Not sure logging to the console would do much good, unless users actually inform us.

Last thought: could we address this on separate PR?